cli/internal/cloudcmd/create.go

```diff
@@ -212,25 +213,35 @@ func (c *Creator) createGCP(ctx context.Context, cl terraformClient, opts Create
 
 func (c *Creator) createAzure(ctx context.Context, cl terraformClient, opts CreateOptions) (idFile clusterid.File, retErr error) {
 	vars := terraform.AzureClusterVariables{
-		CommonVariables: terraform.CommonVariables{
-			Name:               opts.Config.Name,
-			CountControlPlanes: opts.ControlPlaneCount,
-			CountWorkers:       opts.WorkerCount,
-			StateDiskSizeGB:    opts.Config.StateDiskSizeGB,
+		Name: opts.Config.Name,
+		NodeGroups: map[string]terraform.AzureNodeGroup{
+			"control_plane_default": {
+				Role:          "control-plane",
```

Comment:
```suggestion
				Role:          "ControlPlane",
```

Reply:
I was wondering why we have camelCase if we really need dash notation in TF. Any reasoning beside it being Go convention?